### PR TITLE
minor fixes for cheevos-new

### PR DESCRIPTION
**Pull request by meleu**
## Description

This PR fixes two minor issues reported by the user @LootusMaximus while playing with the new cheevos implementation:

1. The achievements feature (and only it) was being completely disabled when a game has a buggy leaderboard entry (and it was happening with the [Pokemon Prism (GBC)](https://retroachievements.org/game/9491) game.
2. the "Pause Hardcore Mode" feature was resetting the game, instead of "pausing" the hardcore mode and continuing the game from the same point, but in softcore mode.

## Related Issues

see above :point_up:

## Related Pull Requests

#8501 

## Reviewers

@leiradel @Jamiras

### CODE REVIEW

**Comment by leiradel** [comment]
Hm, I was going to say that this check in `lboard.c` should be done in the caller:

```c
if (!self) return RC_LBOARD_INACTIVE;
```

We should never slow down the callee with with those kind of checks, if the caller can guarantee the pointer is not `NULL` the check is unnecessary, if it can't it should do the checking itself. If the check stays in `rc_evaluate_lboard` everyone pays the price.

**Comment by twinaphex** [comment]
OK, you want me to revert the PR then so that we can do this over? Sorry for jumping the gun.

**Comment by leiradel** [comment]
No, I don't think it's necessary. It's a minor thing I admit, so it can be done in a different PR.

File: cheevos-new/cheevos.c [hunk 1]
```diff
@@ -380,11 +382,18 @@ static int rcheevos_parse(const char* json)
...
       res = rc_lboard_size(lboard->info->mem);
 
+      CHEEVOS_LOG(RCHEEVOS_TAG "rc_lboard_size() = %d\n", res);
+
       if (res < 0)
       {
-         CHEEVOS_ERR(RCHEEVOS_TAG "Error in leaderboard mem %s: %s",
-            lboard->info->mem, rcheevos_rc_error(res));
-         goto error;
+         snprintf(buffer, sizeof(buffer), "Error in leaderboard %d \"%s\"", lboard->info->id, lboard->info->title);
```
Comment (Jamiras) [inline cheevos-new/cheevos.c]: An average user probably doesn't care either way, but I would have worded this to indicate that it was a Parse Error, probably even including the error code.

File: cheevos-new/cheevos.c [hunk 2]
```diff
@@ -380,11 +382,18 @@ static int rcheevos_parse(const char* json)
       lboard->info = rcheevos_locals.patchdata.lboards + j;
       res = rc_lboard_size(lboard->info->mem);
 
+      CHEEVOS_LOG(RCHEEVOS_TAG "rc_lboard_size() = %d\n", res);
```
Comment (Jamiras) [inline cheevos-new/cheevos.c]: This log message seems unnecessary. We really only care when there's an error, and that's captured on line 394

**Comment by meleu** [comment]
I'm going to submit a new PR later today.

File: cheevos-new/cheevos.c (hunk 2 above)
Comment (meleu) [reply cheevos-new/cheevos.c]: whoops! it's a leftover (used while I was debugging).

File: cheevos-new/cheevos.c (hunk 1 above)
Comment (meleu) [reply cheevos-new/cheevos.c]: oh yeah, I'll add the `rcheevos_rc_error(res)` call here. Even if average users don't know what the message means, it helps when they care enough to report the problem.
thanks for the input :+1:

**Pull request merged**